tt_metal/impl/program/program.hpp

```diff
@@ -131,6 +134,10 @@ class Program {
 
     void allocate_circular_buffers();
 
+    void finalize_rt_args();
+    bool is_finalized() const { return loaded_onto_device; }
+    void set_finalized() { loaded_onto_device = true; }
```

Comment:
Can you change `program.loaded_onto_device = true;` from `command_queue.cpp` around line 1200 to use this api?

Reply:
will do